RELEASE/scripts/autoscend/auto_combat.ash

```diff
@@ -745,26 +763,42 @@ string auto_combatHandler(int round, string opp, string text)
 		}
 	}
 
-	if(thunderBirdsLeft > 0)
+	//Heavy Rain Boss debuffing
+	if($monsters[Big Wisnaqua, The Aquaman, The Big Wisniewski, The Man, The Rain King] contains enemy)
 	{
-		thunderBirdsLeft = thunderBirdsLeft - 1;
-		set_property("auto_combatHandlerThunderBird", thunderBirdsLeft.to_string());
-		return useSkill($skill[Thunder Bird], false);
-	}
-
-	if(canUse($skill[Thunder Bird]) && (monster_level_adjustment() <= 150))
-	{
-		if($monsters[Big Wisnaqua, The Aquaman, The Big Wisniewski, The Man, The Rain King] contains enemy)
+		//Against bosses set how many [Thunder Bird] we want to cast during round 1
+		if(round == 1 && get_property("auto_combatHandlerThunderBird").to_int() == 0)
 		{
-			if(canUse($skill[Curse Of Weaksauce]) && (my_mp() >= 60) && auto_have_skill($skill[Itchy Curse Finger]))
+			int targetThunderBird = 3;
+			if(monster_level_adjustment() > 80)
 			{
-				return useSkill($skill[Curse Of Weaksauce]);
+				targetThunderBird++;
 			}
-
-			set_property("auto_combatHandlerThunderBird", "5");
-			return useSkill($skill[Thunderstrike]);
+			if(monster_level_adjustment() > 110)
+			{
+				targetThunderBird++;
+			}
+			if(monster_level_adjustment() > 150)
+			{
+				targetThunderBird++;
+			}
+			targetThunderBird = min(my_thunder(), targetThunderBird);
+			set_property("auto_combatHandlerThunderBird", targetThunderBird);
+		}
+	
+		//if enemy ML is too high then they cannot be staggered, and at that point curse of weaksauce is less useful as it reduces by a fixed amount.
+		if(canUse($skill[Curse Of Weaksauce]) && my_mp() >= 30 && auto_have_skill($skill[Itchy Curse Finger]) && monster_level_adjustment() <= 150)
+		{
+			return useSkill($skill[Curse Of Weaksauce]);
 		}
 	}
```

Comment:
I think you're missing a `return useSkill($skill[Thunderstrike]);` here.

Reply:
that is handled 2 lines down (also in green because I both relocated it and rewrote it) in the function dedicated to casting thunderbird x number of times being immediately underneath it (2 lines below at line 796).

if I wanted to add a that line where you suggested (where it originally resided) I would have to add a bunch of tests to ensure it goes through properly. which would add up to basically being the entire function and just create a bunch of duplicate code. Initially I did do it but then I realized that by flipping the order on those two I can eliminate that duplicate code.

... i should make a note of that in the comments actually

Reply:
added comments explaning it.
also btw I actually did encounter such an infinite loop before where it just kept on trying to cast an unavailable spell on turn 1 repeatedly.

The reason being that useSkill is a misnomer, it actually returns a string with the correct formatting of the skill to be used instead of being a boolean that tries to use a skill and then return true if it succeeds.

Reply:
[Thunderstrike](https://kol.coldfront.net/thekolwiki/index.php/Thunderstrike) is a different skill from [Thunder Bird](https://kol.coldfront.net/thekolwiki/index.php/Thunder_Bird).

Reply:
> The reason being that useSkill is a misnomer, it does not use a skill, instead it returns a **string** with the correct formatting of the skill to be used instead of being a boolean that tries to use a skill and then return true if it succeeds.

`useSkill()` is not a misnomer. That's how combat consulting works. Your function has to return the action you want used in that round as it is called for every round until the combat ends. This is why we have `canUse()` and why we track skills that can only be used once per combat.

Reply:
I get that this is how combat consulting works. I did not mean it is a misnomer in the sense that it does not actually work.
I meant that the name it has does not indicate exactly what it does
From the name "useSkill" I thought it mean "actually use this skill". Instead what it does is "formatSkillForUse"

if you typed
`useSkill(something);`
it does nothing

instead what it does is format a skill so that it can be used if you call
`return useSkill(something);` 
in combat then it will actually be used

Reply:
> [Thunderstrike](https://kol.coldfront.net/thekolwiki/index.php/Thunderstrike) is a different skill from [Thunder Bird](https://kol.coldfront.net/thekolwiki/index.php/Thunder_Bird).

Oh oops. now what you said makes sense. Alright. I will add it in

Reply:
commit 17ca00a